setup.py

```diff
@@ -26,7 +26,16 @@ def read_file(filename):
     package_data={},
     python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*",
     setup_requires=["setuptools_scm"],
-    install_requires=["click>=6", "six"],
+    install_requires=[
+        # direct dependencies
+        "click>=6",
+        "pip",
+        "six",
+        # indirect dependencies
+        "setuptools",  # typically needed when invoking setup.py
+        "wheel",  # pip plugin needed by pip-tools
+    ],
+    tests_require=["mock", "pytest!=5.1.2"],
```

Comment:
This never got standardized and so I'd prefer this to be configured on the level of tox and/or in-repo constraints files integrated into tox envs.

Reply:
@webknjaz , ok, so remove from here?  I really want to get this PR over with, so I am happy to remove this if that gets it over the line.

To push back a little, `tests_require` has been standardised for decades, and is used by many tools.  For example distro packaging tools read it in order to determine test dependencies needed for their build scripts.

e.g. openSUSE https://github.com/openSUSE/py2pack/ uses metaextract, which uses this field at https://github.com/toabctl/metaextract/blob/a03dd30/metaextract/metaextract.py#L49

All those tools are not going to use tox.ini which is easy to parse and has a stable filename, but impossible to extract meaning from because each project uses it slightly differently; and constraint/requirement files are even worse as different projects use different filenames, and they usually contain `-r` to load other files.  And projects usually forget to include tox.ini or requirement files into their sdists uploaded to PyPI.

Reply:
I'd remove it. Most OS packagers still apply patches and download from Git. Also, PyPA's been actively discouraging using this for at least several years. By not standardized I mean that there's no standard, many features like this were added organically.

Reply:
Also, having it here would force us to maintain the deps list in multiple places.

Reply:
Done

Reply:
ping @atugushev @webknjaz @codingjoe